k8s.io/deployment-short-urls.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: short-urls
+spec:
+  selector:
+    matchLabels:
+      app: short-urls
+  template:
+    metadata:
+      labels:
+        app: short-urls
+    spec:
+      containers:
+      - name: shlink
+        image: shlinkio/shlink:1.16.2
+        command: ["/bin/sh", "-c"]
+        args:
+        - |
+          # Start shlink server in the background.
+          /bin/sh /etc/shlink/docker-entrypoint.sh &
+          PID=$!
+          # Wait for service to become available.
+          while ! shlink short-url:list; do
+            sleep 10s
+          done
+          # Load URLs from the config map.
+          while read slug url; do
+            shlink short-url:generate "--customSlug=${slug}" "${url}"
+          done < /etc/config/short-urls
+          # Create healthcheck URL for our readiness probe.
+          shlink short-url:generate -c shlink-healthcheck https://k8s.io
+          # Block until the config map changes, then exit. This will restart
+          # the container and load the new config.
+          # TODO: live reload without downtime
+          inotifyd - /etc/config/short-urls:e
```

Comment:
What happens if shlink dies for some reason in the background?

Reply:
the container stops serving but stays up until the next config map update. :shame:

Reply:
ooooops..  https://media0.giphy.com/media/9wGjpl8z1CJ5m/giphy.gif?cid=790b76115c9ea09d716e42364db94eed